compaction/task_manager_module.cc

```diff
@@ -467,7 +467,11 @@ future<> shard_cleanup_keyspace_compaction_task_impl::run() {
 
 future<> table_cleanup_keyspace_compaction_task_impl::run() {
     co_await wait_for_your_turn(_cv, _current_task, _status.id);
-    auto owned_ranges_ptr = compaction::make_owned_ranges_ptr(_db.get_keyspace_local_ranges(_status.keyspace));
+    auto get_owned_ranges = [&] (std::string_view keyspace_name) -> owned_ranges_ptr {
+        const auto& erm = _db.find_keyspace(keyspace_name).get_vnode_effective_replication_map();
+        return compaction::make_owned_ranges_ptr(_db.get_keyspace_local_ranges(erm));
+    };
```

Comment:
Isn't the result immediately invalidated by a following range movement?

Reply:
I guess you can say with vnodes the operator is responsible for not performing additional incompatible range movements, but a comment is needed when violating the rules of using effective replication maps.

Reply:
yes, that makes sense.

Reply:
Add comment in v4